components/ntp_background_images/browser/ntp_background_images_source.cc

```diff
@@ -56,7 +61,14 @@ void NTPBackgroundImagesSource::StartDataRequest(
     return;
   }
 
-  auto* images_data = service_->GetBackgroundImagesData();
+  // Favicon data is fetched from cached folder not from component data.
+  if (IsTopSiteFaviconPath(path)) {
+    GetImageFile(GetTopSiteFaviconFilePath(path), std::move(callback));
+    return;
+  }
+
+  auto* images_data =
+      service_->GetBackgroundImagesData(IsSuperReferralPath(path));
```

Comment:
the point of this is that we're not supposed to care if it's a sponsored image or super referral. We only care about the metadata information associated with the image

Reply:
Checking whether `path` SR or SI is added because both SI and SR components are registered simultaneously. In the early implementation, browser only one component is registered. But, it's changed because user can choose `Brave default images` option in the settings. If user choose it, NTP displays SI images also with default BGs even if this is super referrals install.
If user chooses `Super referral` options, NTP only shows SR images.
Because of this change, `NTPBackgroundImagesSource` have to request which component is needed for `path`. From the `NTPBackgroundImagesSource` and `NTPBackgroundImagesService` point, they don't know which is active because it's profile specific.